File: Mathématiques/2 – Quelques outils mathématiques/04 - Approvisionnement de la ruche.py
from math import *
def distance(x1, y1, x2, y2):
    return sqrt(pow(x2 - x1, 2) + pow (y2 - y1, 2))
x = int(input())
y = int(input())
nbFleurs = int(input())
c = 0
for i in range(nbFleurs):
    _x = int(input())
    _y = int(input())
    if(distance(x, y, _x, _y) < 1000):
        c+=1
print(c)

File: Mathématiques/2 – Quelques outils mathématiques/04 - Approvisionnement de la ruche.cpp
#include <iostream>
#include <cmath>
using namespace std;

#define distance sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2))
int main()
{
  int x1, y1, nbFleurs, count = 0;
  cin >> x1 >> y1 >> nbFleurs;
  for (int i = 0; i < nbFleurs; i++)
  {
    int x2, y2;
    cin >> x2 >> y2;
    if (distance < 1000)
    {
      count++;
    }
  }
  cout << count;
}